### ISSUE TITLE
Underscore in links not shown

### ISSUE THREAD

**Issue by besi** [MacDownApp/macdown#1265]
Hy there. I have a link that is not properly displayed in Macdown:

**Expected**: [ble_simple_peripheral.py](https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_simple_peripheral.py)
**Actual**: [blesimpleperipheral.py](https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_simple_peripheral.py)

<img width="832" alt="Screenshot 2021-09-03 at 11 21 27" src="https://user-images.githubusercontent.com/160412/131982488-b35ff0ef-2f83-46fa-8bba-5f7eb3994155.png">

This is my code but the underscores aren't shown in the Preview pane.
```md
Start with [ble_simple_peripheral.py](https://github.com/xyz.py)
```

<img width="372" alt="Screenshot 2021-09-03 at 11 23 48" src="https://user-images.githubusercontent.com/160412/131982750-fad7b65e-b375-4bb3-b98b-e2ce536efaf6.png">

### Update

The problem occurs with all rendering themes. However, when changing the theme I could see that `simple` was actually underlined. So it seems that somehow the underscores are part of the link.

<img width="832" alt="Screenshot 2021-09-03 at 11 30 14" src="https://user-images.githubusercontent.com/160412/131983751-3effc02f-a2bd-4fb7-8cdb-39692b73535d.png">

So I now understand what happened. The underscores got changed into an underline-formatting. I guess this can be seen as expected behaviour.

**Comment by besi** [comment]
So the workaround would be the following. I am wondering if there is a non-code block way to display actual underlines in a given text.

<img width="832" alt="Screenshot 2021-09-03 at 11 32 38" src="https://user-images.githubusercontent.com/160412/131984069-818d9c3f-05cc-4c68-bd8f-f6da094c6a3e.png">

**Comment by arvindh-manian** [comment]
Yes, I believe this to be intended behavior. This can be avoided by escaping the underscores with backslashes.
E.g., you would change the markdown to this
`Start with [ble\_simple\_peripheral.py](https://github.com/micropython/micropython/blob/master/examples/bluetooth/ble_simple_peripheral.py)`
